database/dtb.ksy

```diff
@@ -0,0 +1,94 @@
+meta:
+  id: dtb
+  title: Flattened Devicetree Format
+  file-extension: dtb
+  license: CC0-1.0
+  endian: be
+doc: |
+  Also referred to as Devicetree Blob (DTB). It is a flat
+  binary encoding of data (primarily devicetree data, although
+  other data is possible as well).
+doc-ref:
+  - https://github.com/devicetree-org/devicetree-specification/releases/download/v0.3/devicetree-specification-v0.3.pdf
+  - https://elinux.org/images/f/f4/Elc2013_Fernandes.pdf
+seq:
+  - id: magic
+    contents: [0xd0, 0x0d, 0xfe, 0xed]
+  - id: total_size
+    type: u4
+  - id: structure_block_offset
+    type: u4
+  - id: strings_block_offset
+    type: u4
+  - id: memory_reservation_block_offset
+    type: u4
+  - id: version
+    type: u4
+  - id: last_compatible_version
+    type: u4
+  - id: boot_cpuid_phys
+    type: u4
+  - id: size_dt_strings
+    type: u4
+  - id: size_dt_struct
+    type: u4
+instances:
+  memory_reservation_block:
+    pos: memory_reservation_block_offset
+    size: structure_block_offset - memory_reservation_block_offset
+  structure_block:
+    pos: structure_block_offset
+    size: strings_block_offset - structure_block_offset
+    type: fdt_block
+  strings_block:
+    pos: strings_block_offset
+    size: size_dt_strings
+    type: strings
+types:
+  strings:
+    seq:
+      - id: strings
+        type: strz
+        encoding: ASCII
```

Comment:
citation needed. There is no sense assume anything except utf-8 in 2021.

Reply:
https://github.com/devicetree-org/devicetree-specification/blob/master/source/chapter2-devicetree-basics.rst

It is actually stricter than ASCII:
Valid characters for node names 

Character | Description
-- | --
0-9 | digit
a-z | lowercase letter
A-Z | uppercase letter
, | comma
. | period
_ | underscore
+ | plus sign
- | dash

Reply:
Thanks. Can this be added into the spec?